### PR TITLE
Introduce `run` resource trigger endpoint with zip/json support

### LINKED ISSUE #7: Introduce `run` resource trigger endpoint with zip/json support
Currently, the Bubbly Worker requires that its pool of resource runs are obtained in a push-based approach from the Bubbly Store. Concretely:

1. The Store receives a new/updated `run` resource kind (of _remote_ type), and updates the `_resource` table of the Store's underlying database accordingly.
2. Updates of this type trigger a NATS publication to the Worker telling it to run the new/updated `run` resource

This is all well and good, but makes the assumption that the data required by the Worker to run the resource can be pulled externally (or is available on the worker's filesystem). In some cases, we want to be able to POST _data_, not resources, and have the Worker use this data as input(s) to its `run` resources.

Initial Desired Workflow:

1. JSON/zip data is `POST`ed to the Bubbly API Server, with a parameter identifying the `run` resource.
2. API Server sends the data via a NATS Publication to a Bubbly Worker
3. Bubbly Worker requests the provided `run` resource from the Store via NATS request
4. Bubbly Worker processes the data
5. Bubbly Worker triggers the `run` resource, using the processed data as inputs

Tasks:

- [ ] New API Server endpoint for receiving and shipping the data to the Bubbly Worker. @nate-droid any recommendations for this API endpoint?
- [ ] New Bubbly Worker NATS handler for receiving and processing data, then triggering a `one-off` run of the provided `run` resource using this data as resource `inputs`

@olliefr , @jlarfors: requesting some feedback on this spec as you are more in-tune with the end-user requirements. Any thoughts before I implement 😃 ?

### PR DESCRIPTION
**Pull request by benmarsden**
This PR adds the functionality outlined in issue #7. Namely:

1. introduces the `/api/v1/run/:name` endpoint to the Bubbly API Server
1.1 this endpoint received a named `run` resource and optional input data (`.zip` or `.json`) which the Worker should use when running the named `run` resource
1.2 see `website/docs/resources/kinds.md` for documentation on this endpoint
1.3 see `integration/apply_test.go` for integration test
2. Simplifies the Worker's run logic: now, it maintains a separate _one-off_ pool. The `runs` within this pool are run sequentially whenever the worker receives a new run request from NATS. IMO the disadvantage of sequential vs concurrent is outweighed by the simplicity of the implementation, and in the _one-off_ runs case shouldn't have big performance implications as this pool should never be too large.
2. Explicitly disables `interval` run types in both implementation and documentation. Instead, any `interval` runs are automatically treated as _one-off_. 

closes #7 .

### CODE REVIEW

**Review by jlarfors — APPROVED** [review]
Awesome work, and especially nice that you updated the docs as well. I have not dug deep into the go code, but on a high-level this looks great and I approve!

File: interval/json.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	}
+	defer dst.Close()
+
+	// write new content to the file
+	_, err = dst.Write(b)
+
+	if err != nil {
+		return "", err
+	}
+
```
Comment (olliefr) [inline interval/json.go:43]: I'd close here,

```go
err = dst.Close()

if err != nil {
  return "", err
}
```

File: integration/apply_test.go [hunk 2]
```diff
@@ -124,4 +130,180 @@ func TestApplyRun(t *testing.T) {
...
+	t.Helper()
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+	// part, err := writer.CreateFormFile(paramName, filepath.Base(path))
```
Comment (olliefr) [inline integration/apply_test.go:276]: Commented-out code)

File: interval/zip.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		if err != nil {
+			return err
+		}
+		defer fileReader.Close()
+
+		targetFile, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, file.Mode())
+		if err != nil {
+			return err
+		}
+		defer targetFile.Close()
```
Comment (olliefr) [inline interval/zip.go:67]: I wouldn't `defer` Close on a writeable file, please see my other comment.

**Review by olliefr — CHANGES_REQUESTED** [review]
Wow! 🙃 My only request is to reconsider `defer`ing `Close` on writeable files, as this is not safe.

**Review by nate-droid — APPROVED** [review]
Looks good to me! Thanks for remembering the Swagger Annotations! :)

**Review by olliefr — APPROVED** [review]

**Pull request merged**